sdk/state/ingest_test.go

```diff
@@ -64,16 +64,142 @@ func TestChannel_IngestTx_updateUnauthorizedCloseAgreement(t *testing.T) {
 	assert.Equal(t, int64(0), initiatorChannel.Balance())
 	assert.Equal(t, int64(8), responderChannel.Balance())
 
-	// Pretend that responder broadcasts the declaration tx from the authorized
-	// agreement to the network, making it visible to initiator.
+	// Pretend that responder broadcasts the declaration tx before returning
+	// it to the initiator.
 	declTx, _, err := responderChannel.CloseTxs()
 	require.NoError(t, err)
 	err = initiatorChannel.IngestTx(declTx)
 	require.NoError(t, err)
+	closeState, err := initiatorChannel.CloseState()
+	require.NoError(t, err)
+	require.Equal(t, CloseStateClosing, closeState)
 
 	// The initiator channel and responder channel should have the same close
 	// agreements.
 	assert.Equal(t, int64(8), initiatorChannel.Balance())
 	assert.Equal(t, int64(8), responderChannel.Balance())
 	assert.Equal(t, initiatorChannel.LatestCloseAgreement(), responderChannel.LatestCloseAgreement())
+
+	// TODO - initiatorChannel should be in a close state now, so shouldn't be able to propose/confirm new payments
```

Comment:
I don't think this TODO is correct, ingesting the declTx shouldn't result in the channel being in a close state, it'll still be in a closing state. It won't enter a closed state until the close tx is submitted and ingested.

Reply:
Actually you're right, it should still prevent propose and confirm, but it'll be in a closing state. I misread `be in a close state now` as saying `closed state`.